lib/fizzy/execute.cpp

```diff
@@ -1586,4 +1595,21 @@ ExecutionResult execute(
 trap:
     return Trap;
 }
+}  // namespace
+
+ExecutionResult execute(
+    Instance& instance, FuncIdx func_idx, const Value* args, ExecutionContext& ctx) noexcept
+{
+    if (ctx.metering_enabled)
+        return execute_local_function<true>(instance, func_idx, args, ctx);
+    else
+        return execute_local_function<false>(instance, func_idx, args, ctx);
+}
+
+ExecutionResult execute(Instance& instance, FuncIdx func_idx, const Value* args) noexcept
```

Comment:
I don't think this helps in any way, my main concern was avoiding branching within `invoke_function`'s call to `execute`, which is still happening because that is the `ExecutionContext` overload.